components/livekit/tests/livekit-component.spec.ts

```diff
@@ -0,0 +1,643 @@
+        ...settings,
+        host: 'livekit.example.com'
+      }
+
+      const comp = await createLivekitComponent({ logs }, { settings: settingsWithoutProtocol })
+      const credentials = await comp.generateCredentials({
+        identity: 'test',
+        roomName: 'test-room'
+      })
+
+      expect(credentials.url).toBe('wss://livekit.example.com')
+    })
+  })
+
+  describe('and the host URL already has wss:// prefix', () => {
+    it('should keep the host URL as-is', async () => {
+      const credentials = await component.generateCredentials({
+        identity: 'test',
+        roomName: 'test-room'
+      })
+
+      expect(credentials.url).toBe('wss://livekit.example.com')
+    })
+  })
+})
+
+describe('when generating credentials', () => {
+  describe('and using default permissions', () => {
+    it('should generate credentials with url and token', async () => {
+      const credentials = await component.generateCredentials({
+        identity: 'user-123',
+        roomName: 'my-room'
+      })
+
+      expect(credentials).toEqual({
+        url: 'wss://livekit.example.com',
+        token: 'mock-jwt-token'
+      })
+    })
+  })
+
+  describe('and using custom permissions', () => {
+    it('should generate credentials with the provided permissions', async () => {
+      const credentials = await component.generateCredentials({
+        identity: 'user-123',
+        roomName: 'my-room',
+        permissions: {
+          canPublish: false,
+          canSubscribe: true,
+          canPublishData: false
+        }
+      })
+
+      expect(credentials.token).toBe('mock-jwt-token')
+    })
+  })
+
+  describe('and providing metadata', () => {
+    it('should generate credentials with the provided metadata', async () => {
+      const credentials = await component.generateCredentials({
+        identity: 'user-123',
+        roomName: 'my-room',
+        metadata: { displayName: 'John Doe', role: 'moderator' }
+      })
+
+      expect(credentials.token).toBe('mock-jwt-token')
+    })
+  })
+
+  describe('and providing a custom TTL', () => {
+    it('should generate credentials with the provided TTL', async () => {
+      const credentials = await component.generateCredentials({
+        identity: 'user-123',
+        roomName: 'my-room',
+        ttlSeconds: 600
+      })
+
+      expect(credentials.token).toBe('mock-jwt-token')
+    })
+  })
+})
+
+describe('when building connection URL', () => {
+  it('should build a connection URL from host and token', () => {
+    const url = component.buildConnectionUrl('wss://livekit.example.com', 'my-token')
+    expect(url).toBe('livekit:wss://livekit.example.com?access_token=my-token')
+  })
+})
+
+describe('when listing rooms', () => {
+  describe('and no filter is provided', () => {
+    it('should list all rooms', async () => {
+      const mockRooms = [{ name: 'room-1' }, { name: 'room-2' }]
+      mockRoomClient.listRooms.mockResolvedValue(mockRooms)
```

Comment:
Would you mind moving the mocks to a `beforeEach`? The same would apply to other tests in the file.